gcs-action/src/gcs.ts

```
@@ -0,0 +1,202 @@
+ * based on https://github.com/googleapis/nodejs-storage/blob/master/samples/listFilesByPrefix.js
+ * @param bucketName The name of the bucket
+ * @param prefix The prefix to filter by
+ * @param delimiter Optional delimiter
+ */
+export async function listFilesByPrefix(
+  bucketName: string,
+  prefix: string,
+  delimiter?: string
+): Promise<File[]> {
+  //interface for listing options
+  interface ListOptions {
+    prefix: string
+    delimiter?: string
+  }
+  const options: ListOptions = {prefix}
+
+  if (delimiter) {
+    options.delimiter = delimiter
+  }
+  // Lists files in the bucket, filtered by a prefix
+  const [files] = await storage.bucket(bucketName).getFiles(options)
+  return files
+}
+
+/**
+ * Deletes a file in a bucket
+ * based on https://github.com/googleapis/nodejs-storage/blob/master/samples/deleteFile.js
+ * @param bucketName The name of the bucket
+ * @param filename The name of the name of the file to be deleted
+ */
+export async function deleteFile(
+  bucketName: string,
+  filename: string
+): Promise<{status: DeleteFileResponse}> {
+  try {
+    const deletedFile = await storage
+      .bucket(bucketName)
+      .file(filename)
+      .delete()
+    console.log(`gs://${bucketName}/${filename} deleted.`)
+    return {status: deletedFile}
+  } catch (err) {
+    return {status: err}
+  }
+}
+
+/**
+ * Uploads a specified directory to a GCS bucket
+ * based on https://github.com/googleapis/nodejs-storage/blob/master/samples/uploadDirectory.js
+ * @param bucketName The name of the bucket
+ * @param directoryPath The path of the directory to upload
+ * @param objectKeyPrefix Optional Prefix for in the GCS bucket
+ * @param clearExistingFilesFirst Clean files in the prefix before uploading
+ */
+export async function uploadDirectory(
+  bucketName: string,
+  directoryPath: string,
+  objectKeyPrefix: string = '',
+  clearExistingFilesFirst: boolean = false
+): Promise<{fileName: string; status: UploadResponse | Error}[] | undefined> {
```

Just return a filename, let the error bubble.

---

Done